lib/orderbook/OrderBook.ts

```diff
@@ -357,26 +357,32 @@ class OrderBook extends EventEmitter {
           return;
         }
 
-        try {
-          this.logger.debug(`matched with peer ${maker.peerPubKey}, executing swap on taker ${taker.id} and maker ${maker.id} for ${maker.quantity}`);
-          const swapSuccess = await this.executeSwap(maker, taker);
-
-          if (swapSuccess.quantity < maker.quantity) {
+        this.logger.debug(`matched with peer ${maker.peerPubKey}, executing swap on taker ${taker.id} and maker ${maker.id} for ${maker.quantity}`);
+        const swapResult = await this.executeSwap(maker, taker);
+        if (isSwapSuccess(swapResult)) {
+          if (swapResult.quantity < maker.quantity) {
             // swap was only partially completed
-            portion.quantity = swapSuccess.quantity;
-            const rejectedQuantity = maker.quantity - swapSuccess.quantity;
-            this.logger.info(`match partially executed on taker ${taker.id} and maker ${maker.id} for ${swapSuccess.quantity} ` +
+            portion.quantity = swapResult.quantity;
+            const rejectedQuantity = maker.quantity - swapResult.quantity;
+            this.logger.info(`match partially executed on taker ${taker.id} and maker ${maker.id} for ${swapResult.quantity} ` +
               `with peer ${maker.peerPubKey}, ${rejectedQuantity} quantity not accepted and will repeat matching routine`);
             await retryFailedSwap(rejectedQuantity);
           } else {
             this.logger.info(`match executed on taker ${taker.id} and maker ${maker.id} for ${maker.quantity} with peer ${maker.peerPubKey}`);
           }
-          swapSuccesses.push(swapSuccess);
-          onUpdate && onUpdate({ type: PlaceOrderEventType.SwapSuccess, payload: swapSuccess });
-        } catch (err) {
+          swapSuccesses.push(swapResult);
+          onUpdate && onUpdate({ type: PlaceOrderEventType.SwapSuccess, payload: swapResult });
+        } else {
           this.logger.warn(`swap for ${portion.quantity} failed during order matching, will repeat matching routine for failed swap quantity`);
```

Comment:
You can log here the swap failure reason as well (same as you did in `OrderBook.executeSwap`.

Reply:
Wouldn't that be repetitive? Since this is calling `OrderBook.executeSwap` I figure it would log the same error twice and both from the `ORDERBOOK` logger. I wanted to have the error logging within `executeSwap` so that we get similar logging in `nomatching` mode as well. If you still see a reason to have it here as well let me know and I'll add it.

Reply:
I didn't notice it's calling `OrderBook.executeSwap`. Forget about it!